src/main/java/net/minestom/server/extensions/ExtensionManager.java

```diff
@@ -612,10 +612,23 @@ public Extension getExtension(@NotNull String name) {
         return extensions.get(name.toLowerCase());
     }
 
+    @Nullable
+    @SuppressWarnings("unchecked")
+    public <E> E getExtension(@NotNull Class<E> type) {
+        return (E) extensions.values().stream()
+                .filter(extension -> extension.getClass().isAssignableFrom(type))
```

Comment:
This won't work if different extensions try to access the same types: 
Extension A will have a different value when loading 'TestExtension.class' than the value Extension B will have.